File: Tier2/src/main/java/via/sep3/group11/tier2/logicInterfaces/AuthInterface.java
package via.sep3.group11.tier2.logicInterfaces;

import via.sep3.group11.tier2.shared.DTOs.*;

public interface AuthInterface {
    RefugeeDTO registerRefugee(NewRefugeeRegisterDTO dto);
    HostDTO registerHost(NewHostRegisterDTO dto);
    boolean existsByEmail(String email);
    AuthResponseDTO login(LoginDTO loginDTO);
}

File: Tier1/HttpClients/ClientInterfaces/AuthInterface.cs
using System.Security.Claims;
using Shared.DTOs;

namespace HttpClients.ClientInterfaces;

public interface AuthInterface
{
    public Task<AuthResponseDTO> LoginAsync(LoginDTO dto);
    public Task LogoutAsync();
    public Task<HostDTO> RegisterHostAsync(HostRegisterDTO dto);
    public Task<RefugeeDTO> RegisterRefugeeAsync(RefugeeRegisterDTO dto);
    public Task<ClaimsPrincipal> GetAuthAsync();
    public string GetEmail();

    public Action<ClaimsPrincipal> OnAuthStateChanged { get; set; }
}